File: java/src/main/java/vsdk/toolkit/raycasting/photonMap/PhotonMapConfig.java
/**
Photon map configuration structure, used during construction
*/

package vsdk.toolkit.raycasting.photonMap;

import vsdk.toolkit.raycasting.bidirectionalRaytracing.BiPath;
import vsdk.toolkit.raycasting.bidirectionalRaytracing.LightList;
import vsdk.toolkit.raycasting.raytracing.SamplerConfig;
import vsdk.toolkit.render.ScreenBuffer;

public class PhotonMapConfig {
    public SamplerConfig lightConfig;
    public SamplerConfig eyeConfig;
    public BiPath biPath;

    public ImportanceMap importanceMap;
    public ImportanceMap importanceCMap;
    public PhotonMap map;
    public PhotonMap causticMap;

    public PhotonMap currentMap; // Map in current use: global or caustic
    public ImportanceMap currentImpMap; // Importance Map in current use: global or caustic

    public ScreenBuffer screen;
    public LightList lightList;

    public PhotonMapConfig() {
        lightConfig = new SamplerConfig();
        eyeConfig = new SamplerConfig();
        biPath = new BiPath();

        importanceMap = null;
        importanceCMap = null;
        map = null;
        causticMap = null;

        currentMap = null;
        currentImpMap = null;

        screen = null;
        lightList = null;
    }
}

File: typescript/src/main/vsdk/toolkit/raycasting/photonMap/PhotonMapConfig.ts
/**
Photon map configuration structure, used during construction
*/

import { ScreenBuffer } from "../../render/ScreenBuffer";
import { BiPath } from "../bidirectionalRaytracing/BiPath";
import { LightList } from "../bidirectionalRaytracing/LightList";
import { SamplerConfig } from "../raytracing/SamplerConfig";
import { ImportanceMap } from "./ImportanceMap";
import { PhotonMap } from "./PhotonMap";

export class PhotonMapConfig {
  public lightConfig: SamplerConfig;
  public eyeConfig: SamplerConfig;
  public biPath: BiPath;

  public importanceMap: ImportanceMap | null;
  public importanceCMap: ImportanceMap | null;
  public map: PhotonMap | null;
  public causticMap: PhotonMap | null;

  public currentMap: PhotonMap | null;
  public currentImpMap: ImportanceMap | null;

  public screen: ScreenBuffer | null;
  public lightList: LightList | null;

  public constructor() {
    this.lightConfig = new SamplerConfig();
    this.eyeConfig = new SamplerConfig();
    this.biPath = new BiPath();

    this.importanceMap = null;
    this.importanceCMap = null;
    this.map = null;
    this.causticMap = null;

    this.currentMap = null;
    this.currentImpMap = null;

    this.screen = null;
    this.lightList = null;
  }
}

